File: app/src/main/webapp/js/classes/most-interface.js
class MostInterface {
    static fromJson(json) {
        const mostInterface = new MostInterface();

        mostInterface.setId(json.id);
        mostInterface.setMostId(json.mostId);
        mostInterface.setName(json.name);
        mostInterface.setDescription(json.description);
        mostInterface.setLastModifiedDate(json.lastModifiedDate);
        mostInterface.setReleaseVersion(json.releaseVersion);
        mostInterface.setBaseVersionId(json.baseVersionId);
        mostInterface.setPriorVersionId(json.priorVersionId);
        mostInterface.setIsReleased(json.isReleased);

        return mostInterface;
    }

    static toJson(mostInterface) {
        return {
            id:                 mostInterface.getId(),
            mostId:             mostInterface.getMostId(),
            name:               mostInterface.getName(),
            description:        mostInterface.getDescription(),
            lastModifiedDate:   mostInterface.getLastModifiedDate(),
            releaseVersion:     mostInterface.getReleaseVersion(),
            baseVersionId:      mostInterface.getBaseVersionId(),
            priorVersionId:     mostInterface.getPriorVersionId(),
            isReleased:         mostInterface.isReleased(),
        };
    }

    constructor() {
        this._id                    = null;
        this._mostId                = null;
        this._name                  = "";
        this._description           = "";
        this._lastModifiedDate      = "";
        this._version               = "";
        this._versionsJson          = null;
        this._isReleased            = null;
        this._priorVersionId        = null;
        this._baseVersionId         = null;

        this._functions             = [];
    };

    setId(id) {
        this._id = id;
    }

    getId() {
        return this._id;
    }

    setMostId(mostId) {
        this._mostId = mostId;
    }

    getMostId() {
        return this._mostId;
    }

    setName(name) {
        this._name = name;
    }

    getName() {
        return this._name;
    }

    setDescription(description) {
        this._description = description;
    }

    getDescription() {
        return this._description;
    }

    setLastModifiedDate(lastModifiedDate) {
        this._lastModifiedDate = lastModifiedDate;
    }

    getLastModifiedDate() {
        return this._lastModifiedDate;
    }

    setReleaseVersion(version) {
        this._version = version;
    }

    getReleaseVersion() {
        return this._version;
    }

    getFunctions() {
        return this._functions;
    }

    setVersionsJson(versionsJson) {
        this._versionsJson = versionsJson;
    }

    getVersionsJson() {
        return this._versionsJson;
    }

    setBaseVersionId(baseVersionId) {
        this._baseVersionId = baseVersionId;
    }

    getBaseVersionId() {
        return this._baseVersionId;
    }

    setPriorVersionId(priorVersionId) {
        this._priorVersionId = priorVersionId;
    }

    getPriorVersionId() {
        return this._priorVersionId;
    }

    setIsReleased(isReleased) {
        this._isReleased = isReleased;
    }

    isReleased() {
        return this._isReleased;
    }

    getDisplayVersion() {
        if (this._isReleased) {
            return this._version;
        }
        return this._version + "-" + this._id;
    }
}

File: most-adapter/src/main/java/com/softwareverde/tidyduck/MostInterface.java
package com.softwareverde.tidyduck;

import org.w3c.dom.Document;
import org.w3c.dom.Element;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class MostInterface extends XmlNode {

    private Long _id;
    private String _mostId;
    private String _name;
    private String _description;
    private String _version;
    private Date _lastModifiedDate;
    private boolean _isCommitted;
    private List<Modification> _modifications = new ArrayList<>();
    private List<MostFunction> _mostFunctions = new ArrayList<>();

    public Long getId() {
        return _id;
    }

    public void setId(Long id) {
        _id = id;
    }

    public String getMostId() {
        return _mostId;
    }

    public void setMostId(String mostId) {
        _mostId = mostId;
    }

    public String getName() {
        return _name;
    }

    public void setName(String name) {
        _name = name;
    }

    public String getDescription() {
        return _description;
    }

    public void setDescription(String description) {
        _description = description;
    }

    public String getVersion() {
        return _version;
    }

    public void setVersion(String version) {
        _version = version;
    }

    public Date getLastModifiedDate() {
        return _lastModifiedDate;
    }

    public void setLastModifiedDate(Date lastModifiedDate) {
        _lastModifiedDate = lastModifiedDate;
    }

    public boolean isCommitted() {
        return _isCommitted;
    }

    public void setCommitted(boolean committed) {
        _isCommitted = committed;
    }

    public List<Modification> getModifications() {
        return _modifications;
    }

    public void addModification(final Modification modification) {
        _modifications.add(modification);
    }

    public List<MostFunction> getMostFunctions() {
        return new ArrayList<>(_mostFunctions);
    }

    public void addMostFunction(MostFunction mostFunction) {
        _mostFunctions.add(mostFunction);
    }

    @Override
    public Element generateXmlElement(Document document) {
        throw new UnsupportedOperationException("generateXmlElement is invalid for MostInterface as there is not associated XML for interfaces.");
    }
}
